modules/shortcodes/flickr.php

```diff
@@ -23,95 +23,64 @@
  * @param string $content Post content.
  */
 function flickr_embed_to_shortcode( $content ) {
-	if ( ! is_string( $content ) || false === stripos( $content, '/www.flickr.com/apps/video/stewart.swf' ) ) {
+	if ( ! is_string( $content ) ) {
 		return $content;
 	}
 
-	$regexp     = '%(<object.*?(?:<(?!/?(?:object|embed)\s+).*?)*?)?<embed((?:\s+\w+="[^"]*")*)\s+src="http(?:\:|&#0*58;)//www.flickr.com/apps/video/stewart.swf[^"]*"((?:\s+\w+="[^"]*")*)\s*(?:/>|>\s*</embed>)(?(1)\s*</object>)%';
-	$regexp_ent = str_replace(
-		array(
-			'&amp;#0*58;',
-			'[^&gt;]*',
-			'[^&lt;]*',
-		),
-		array(
-			'&amp;#0*58;|&#0*58;',
-			'[^&]*(?:&(?!gt;)[^&]*)*',
-			'[^&]*(?:&(?!lt;)[^&]*)*',
-		),
-		htmlspecialchars( $regexp, ENT_NOQUOTES )
-	);
+	$type = '';
 
-	foreach ( compact( 'regexp', 'regexp_ent' ) as $reg => $regexp ) {
-		if ( ! preg_match_all( $regexp, $content, $matches, PREG_SET_ORDER ) ) {
-			continue;
-		}
-		foreach ( $matches as $match ) {
-			$params = $match[2] . $match[3];
+	if ( false !== strpos( $content, '<div class="flickr_video"' ) && false !== strpos( $content, '<video' ) ) {
+		$type = 'video';
+	}
 
-			if ( 'regexp_ent' === $reg ) {
-				$params = html_entity_decode( $params );
-			}
+	if ( preg_match( '/<iframe src="(https?:)?\/\/([\da-z\-]+\.)*?((static)?flickr\.com|flic\.kr)\/[^\"]+\"/', $content ) ) {
+		$type = 'photo';
+	}
 
-			$params = wp_kses_hair( $params, array( 'http' ) );
-			if (
-				! isset( $params['type'] )
-				|| 'application/x-shockwave-flash' !== $params['type']['value']
-				|| ! isset( $params['flashvars'] )
-			) {
-				continue;
-			}
+	if ( empty( $type ) ) {
+		return $content;
+	}
 
-			$flashvars = array();
-			wp_parse_str( html_entity_decode( $params['flashvars']['value'] ), $flashvars );
+	if ( 'video' === $type ) {
+		// Get video src.
+		preg_match( '/<video src=\"([^\"]+)\"/', $content, $matches );
 
-			if ( ! isset( $flashvars['photo_id'] ) ) {
-				continue;
-			}
+		if ( empty( $matches[1] ) ) {
+			return $content;
+		}
 
-			$photo_id = preg_replace( '#[^A-Za-z0-9_./@+-]+#', '', $flashvars['photo_id'] );
+		preg_match( '/\d+/', $matches[1], $matches );
 
-			if ( ! strlen( $photo_id ) ) {
-				continue;
-			}
+		$video_id = $matches[0];
 
-			$code_atts = array( 'video' => $photo_id );
+		// Get width.
+		preg_match( '/width:\s(\d+)px/', $content, $matches );
 
-			if (
-				isset( $flashvars['flickr_show_info_box'] )
-				&& 'true' === $flashvars['flickr_show_info_box']
-			) {
-				$code_atts['show_info'] = 'true';
-			}
+		$width = empty( $matches[1] ) ? '' : 'w=' . $matches[1];
```

Comment:
It may be good to escape those matches as we use them.

Reply:
Thanks noticing this!, I added the `esc_attr` calls in my last commit :+1: